pkg/server/server.go

```diff
@@ -491,8 +496,13 @@ func (s *ProxyServer) serveRecvFrontend(stream client.ProxyService_ProxyServer,
 			random := pkt.GetCloseDial().Random
 			klog.V(5).InfoS("Received DIAL_CLOSE", "serverID", s.serverID, "dialID", random)
 			// Currently not worrying about backend as we do not have an established connection,
-			s.PendingDial.Remove(random)
-			klog.V(5).InfoS("Removing pending dial request", "serverID", s.serverID, "dialID", random)
+			if pd := s.PendingDial.Remove(random); pd != nil {
+				klog.ErrorS(nil, "Dial closed prematurely",
```

Comment:
Could it be misleading to consider this an error or should it be V(2) (or even V=0)? My current understanding is that sometimes `DIAL_CLS` could be received before the initial dial completes (e.g. short context deadline, etc)

Reply:
+1. We know pending dials can be removed on one of several paths. This is not necessarily an error.

Reply:
Restating from the PR description:

> On DIAL_CLS, log that the dial was canceled. These are logged at the error level, since it represents a "dial failure", even though it's not technically an error from the server perspective. This is inline with how we log agent errors in the dial response.

IMO logging this at error level is consistent with logging the error in the response at error level (https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/master/pkg/server/server.go#L764). This feels like a good fit for `Warning` level, but it seems like warning level is being deprecated? https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#change-log-functions

I could drop these to an info level, but the way I forsee using these logs is to debug dial failures, and logging them at error level makes the failures standout much more easily (at least in GCP logs).

WDYT?

Reply:
Message changed to `Dial cancelled (DIAL_CLS) by frontend`